x-pack/plugins/apm/public/components/shared/TransactionBreakdown/index.tsx

```diff
@@ -39,14 +30,13 @@ function TransactionBreakdown() {
           </EuiTitle>
         </EuiFlexItem>
         <EuiFlexItem grow={false}>
-          {noHits ? (
-            <EuiText>{emptyMessage}</EuiText>
-          ) : (
-            <TransactionBreakdownKpiList kpis={kpis} />
-          )}
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <TransactionBreakdownGraph timeseries={timeseries} />
+          <TransactionBreakdownGraph
+            timeseries={timeseries.map((serie) => ({
+              ...serie,
+              legendValue: asPercent(serie.legendValue, 1),
```

Comment:
Any reason for overwriting series here, and not where it's defined?
https://github.com/elastic/kibana/blob/63265b6f57e421c335945aa4e948bd3334876222/x-pack/plugins/apm/server/lib/transactions/breakdown/index.ts#L211-L217

Reply:
`asPercent` is defined in `/public` and I think the server doesn't have to know how the UI is going to show the value, it should only provide the value. I have my concerns about `color`, `type` and `hideLegend`, maybe they should all be removed from the server and be defined only in the client.

Reply:
See #70161 in the backlog.

Reply:
> I think the server doesn't have to know how the UI is going to show the value, it should only provide the value.

It depends. Are there any drawbacks to defining it on the server? I see clear drawbacks in defining the chart config across multiple files. 

@smith Regarding #70161: I'd be surprised if we don't have a way to access the advanced settings from the server.

Reply:
Thanks for your input @sqren, the only drawback I can see here is that I'd have to move the `asPercent` to a common place. But if it makes sense the server to send the value already formatted then I wouldn't think it as a drawback but as an improvement. My question is does it make sense sending the value already formatted from the server? For what I've seen we never send any formatted value from the server, we always leave this responsibility to the client. 

FYI: I'm not against moving the `asPercent`, I only want to keep consistency across all APIs we've got.

Reply:
> For what I've seen we never send any formatted value from the server, we always leave this responsibility to the client.

API purists would definitely frown about doing formatting on the server. But I have no shame and am okay with a (rather) close coupling between client and server. I think it's already a reality today.  
What I care most about are the co-location of code where it makes sense. So in this case it feels like an artificial separation to format the legend on the client, while specifying whether the legend should be visible on the server.

Btw. if `hideLegend: true` will `legendValue: <value>` have any effect?

Reply:
I also have no shame but disagree. Presentational elements should be determined on the client. We don't have to fix all of these problems today but should work toward improvement.

Reply:
@smith: why? (Other than it being the convention)

Reply:
On a technical level, some explanation is in #70161. If a users are using different themes and the API includes the colors of the lines the API would have to return different responses based on the theme.

If the presentational elements are coupled with the chart data, changing the presentation becomes more difficult.

Reply:
For theming then it makes sense, although I can imagine that can be supported on the server as well. Personally I'm not swayed by "presentational data belongs on the client". IME, server-side code is more predictable and testable, and it's nice to have it in one place.

Reply:
Sounds good @cauemarcondes 👍